### PR TITLE
sql: new statement `SHOW ENUMS`

### LINKED ISSUE #52328: sql: add a SHOW ENUMS
It'd be nice to easily see all the ENUM types for a cluster. In addition, it'd be great to be able to see them with the enumsortorder and all values. You can construct a query from internal tables but it'd be great if we made an alias for convenience:
```SQL
SELECT
  type.typname AS name,
  string_agg(enum.enumlabel, '|') AS value
FROM pg_enum AS enum
JOIN pg_type AS type
  ON (type.oid = enum.enumtypid)
GROUP BY type.typname;
    name    |                    value
------------+-----------------------------------------------
  color_z   | blue|red|purple|gray|yellow|pink|coal|orange
  weather   | sunny|rainy|cloudy|snow
  transport | bus|tram|rail|ferry
(3 rows)

Time: 9.742ms
```

### PR DESCRIPTION
**Pull request by ekalinin**
Fixes #52328 

Add `SHOW ENUMS` statement listing existing enums.

Release note (sql change): add new statement `SHOW ENUMS` which displays
the information on enums.

### CODE REVIEW

File: pkg/sql/delegate/show_enums.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package delegate
+
+import "github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+
+func (d *delegator) delegateShowEnums(n *tree.ShowEnums) (tree.Statement, error) {
+	query := `
+SELECT type.typname                     AS name,
```
Comment (rohany) [inline pkg/sql/delegate/show_enums.go:17]: Format this with sqlfum.pt please

File: pkg/sql/sem/tree/show.go [hunk 2]
```diff
@@ -137,6 +137,15 @@ func (node *ShowDatabases) Format(ctx *FmtCtx) {
 	}
 }
 
+// ShowEnums represents a SHOW ENUMS statement.
+type ShowEnums struct {
```
Comment (rohany) [inline pkg/sql/sem/tree/show.go:141]: nit just put this on on line `type ShowEnums struct {}`

**Review by rohany — CHANGES_REQUESTED** [review]
Small nits. You're on a roll these days @ekalinin !

File: pkg/sql/delegate/show_enums.go (hunk 1 above)
Comment (ekalinin) [reply pkg/sql/delegate/show_enums.go]: Done

File: pkg/sql/sem/tree/show.go (hunk 2 above)
Comment (ekalinin) [reply pkg/sql/sem/tree/show.go]: Done

**Comment by ekalinin** [comment]
> You're on a roll these days @ekalinin !

Ha-ha, yeah, had some spare time this week.

**Comment by rohany** [comment]
LGTM. Thanks for the contribution!

bors r+

**Pull request merged**